### PR TITLE
182- plural header shows only once when have multiple plural strings

### PR DESCRIPTION
**Pull request by DarjaLeonova**
https://github.com/translate-agent/translate/issues/182

### CODE REVIEW

File: pkg/convert/pot_test.go [hunk 1]
```diff
@@ -708,6 +793,8 @@ msgstr "Au revoir!"
 			result, err := ToPot(tt.input)
 			require.NoError(t, err)
 
+			t.Logf("actual: %v", string(result))
+			t.Logf("expect: %v", string(tt.expected))
```
Comment (VladislavsPerkanuks) [inline pkg/convert/pot_test.go:797]: :)
```suggestion
```

File: pkg/convert/pot_test.go [hunk 2]
```diff
@@ -625,6 +628,88 @@ msgstr[0] ""
...
+						Message:     "match {$count :number}\nwhen 1 {Il y a {$count} pomme.}\nwhen * {Il y a {$count} pommes.}",
+						Description: "apple counts",
+						Status:      model.MessageStatusFuzzy,
+					},
+				},
+			},
+			expected: []byte(`msgid ""
+msgstr ""
+"Language: fr\n"
+#. apple counts
```
Comment (VladislavsPerkanuks) [inline pkg/convert/pot_test.go:700]: missing newline ?
```suggestion
msgstr ""
"Language: fr\n"

#. apple counts
...
```

File: pkg/convert/pot.go [hunk 3]
```diff
@@ -31,6 +33,12 @@ func ToPot(t model.Translation) ([]byte, error) {
 		return nil, fmt.Errorf("write language: %w", err)
 	}
 
+	if !t.Original {
+		if err := writePluralHeader(&b, t); err != nil {
```
Comment (VladislavsPerkanuks) [inline pkg/convert/pot.go:37]: To simplify everything, we can add Plural headers to non-original files, without checking, if they actually have plurals.
e.g.
```suggestion
		fmt.Fprintf(b, "\"Plural-Forms: nplurals=%d; plural=(n != 1);\\n\"\n\n", pluralCountLimit)
```
Somewhere later, we could adjust that header based on language

File: pkg/convert/pot.go [hunk 4]
```diff
@@ -21,6 +21,8 @@ const (
 	MsgStr       poTag = "msgstr"
 )
 
+const pluralCountLimit = 2
```
Comment (janishorsts) [inline pkg/convert/pot.go:24]: In general, keep everything as close as possible! If const is used within a single function, do not declare it at the package level. Declare it inside the function that consumes it.

There are no benefits to declaring it at the package level.

File: pkg/convert/pot.go [hunk 5]
```diff
@@ -40,10 +48,22 @@ func ToPot(t model.Translation) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
+func writePluralHeader(b *bytes.Buffer, t model.Translation) error {
```
Comment (janishorsts) [inline pkg/convert/pot.go:51]: is the plural header the same for the whole translation file? Please add a description and URL to POT documentation to ease understanding.

File: pkg/convert/pot.go [hunk 6]
```diff
@@ -40,10 +48,22 @@ func ToPot(t model.Translation) ([]byte, error) {
...
+	for _, message := range t.Messages {
+		if message.PluralID != "" {
+			if _, err := fmt.Fprintf(b, "\"Plural-Forms: nplurals=%d; plural=(n != 1);\\n\"\n\n", pluralCountLimit); err != nil { //nolint:lll
+				return fmt.Errorf("write plural forms: %w", err)
+			}
+
+			break
```
Comment (janishorsts) [inline pkg/convert/pot.go:58]: `return` conveys the meaning and intent better. If you use `break`, it means there is additional processing but there is none actually.

```suggestion
			return nil
```

File: pkg/convert/pot.go (hunk 4 above)
Comment (DarjaLeonova) [reply pkg/convert/pot.go:24]: A constant pluralCountLimit is used in two different functions. (FromPot and ToPot).

**Review by VladislavsPerkanuks — APPROVED** [review]

**Review by janishorsts — APPROVED** [review]

**Pull request merged**